### PR TITLE
BigQuery: Add CopyJob::Updater class.

### PR DESCRIPTION
**Pull request by tswast**
Another step towards #1960

### CODE REVIEW

**Comment by tswast** [comment]
This commit is now at #2016

**Comment by quartzmo** [comment]
The CI build failed for a reason unrelated to this PR that has since been resolved. Restarting.

File: google-cloud-bigquery/lib/google/cloud/bigquery/copy_job.rb [hunk 1]
```diff
@@ -129,6 +129,105 @@ def write_empty?
...
+                copy: Google::Apis::BigqueryV2::JobConfigurationTableCopy.new(
+                  source_table: source,
+                  destination_table: target
+                ),
+                dry_run: options[:dryrun]
+              )
+            )
+
+            updater = CopyJob::Updater.new req
+            updater.create = options[:create]
```
Comment (quartzmo) [inline google-cloud-bigquery/lib/google/cloud/bigquery/copy_job.rb]: Does this need to be converted with ` Convert.create_disposition(options[:create])`?

(Is there an existing test covering these assignments?)

Comment (tswast) [reply google-cloud-bigquery/lib/google/cloud/bigquery/copy_job.rb]: Yes it needs to be converted, but this conversion is done in `def create=`.

Yes, it appears this behavior is tested at https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/f479fbcd3a280df07ec85218a7c98f76c5d40697/google-cloud-bigquery/test/google/cloud/bigquery/table_copy_test.rb#L123 and https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/f479fbcd3a280df07ec85218a7c98f76c5d40697/google-cloud-bigquery/test/google/cloud/bigquery/table_copy_test.rb#L156

Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery/copy_job.rb]: Oh yes of course. I saw that, but then mistakenly thought this assignment was to the gapi object. Thanks.

**Comment by blowmage** [comment]
@tswast There is a conflict now that #2014 is merged. Can you rebase on `bigquery` and fix the conflict? This looks good otherwise.

**Comment by tswast** [comment]
@blowmage Rebased.

**Comment by blowmage** [comment]
FYI, it looks like the multiple merge commits in this branch aren't allowing a clean rebase. We may have to squash this branch when merging.

**Comment by blowmage** [comment]
Ugh, my comment about squashing because of the merge commits was meant for #2016, not for this PR. My apologies.

**Pull request merged**